cpp/include/cudf/detail/reverse.hpp

```diff
@@ -0,0 +1,56 @@
+/*
```

Comment:
I think unless and until the detail APIs are used outside of the implementation (e.g. by another translation unit), they should be kept inside the implementation .cu file. So no need for this header yet.